src/PowerSystem/power_network.jl

```diff
@@ -77,6 +105,12 @@ struct PowerNetwork <: AbstractPowerSystem
 
         # bus type indexing
         ref, pv, pq, bustype = bustypeindex(bus, gen, bus_id_to_indexes)
+        # check consistency
+        ref_id = bus[ref, 1]
+        @assert bus[ref, 2] == [REF_BUS_TYPE]
+        if !(ref_id[1] in gen[:, 1])
+            println("[PS] No generator affected to slack node.")
```

Comment:
@frapac here I would say "attached" in place of "affected". 

Perhaps we can throw an error, I am not sure what behavior we should have in this case.

Reply:
Good catch. "attached" is better than "affected". I will replace the print by throwing an error there.